- [![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/shubhankarreddy/From-Python-to-ML/blob/main/notebooks/01.Core-Python-Foundations/01.Variables-and-Data-Types.ipynb) 01. Variables and Data Types


# **What is a Variable?**

A **variable** is a name that stores a value.

Think of it like a labelled box.

**Example:**

If we write:

age = 25

Python creates a box.
The box name is "age".
Inside the box, it stores the value 25.

In [ ]:
age = 25

In [ ]:
# To print age variable

print(age)

25


**Explanation:**


1.   **age** is the variable name.
2.   **=** is the assignment operator
3.   **25** is the value stored


**print(age)**  tells Python to show the value stored in age variable.

# **Rules for Naming Variables**

1. Must start with a letter or underscore
2. Cannot start with a number
3. No spaces allowed
4. Case sensitive (Age and age are different)
5. Cannot use Python keywords like if, for, while

The following are **INVALID** variable names:

1. 1name = "Rahul"      : It starts with number
2. student name = "A"   :  contains space

In [ ]:
# Example

marks = 90
student_name = "Rahul"
_total = 100

print(marks)
print(student_name)
print(_total)

90
Rahul
100


# **What is a Data Type?**

A data type tells Python what kind of value we are storing.

Common data types:

1. **int**   for whole numbers  
2. **float** for decimal numbers  
3. **str**   for text  
4. **bool**  for True or False values

In [ ]:
x = 10
print(x)
print(type(x))

10
<class 'int'>


**type(x)** tells us the type of data stored in x.

**<class 'int'>** means it is an integer.

In [ ]:
price = 99.99
print(price)
print(type(price))

99.99
<class 'float'>


In [ ]:
name = "Shubhankar"
print(name)
print(type(name))

Shubhankar
<class 'str'>


Text must always be written inside quotes.

**Correct:**
1. "Hello"
2. 'Hello'

**Wrong:**
1. Hello

**If you write Hello without quotes, Python thinks it is a variable.**

# **Dynamic Typing**

Python automatically understands the data type.

We do not need to declare it manually.

This is called **dynamic typing**.

In [ ]:
value = 10
print(type(value))

value = "Hello"
print(type(value))

<class 'int'>
<class 'str'>


In **Data Science**, every dataset has columns.

Each column has a data type.

Example:

1. Age column → int  
2. Salary column → float  
3. Name column → string  
4. Is_Active column → boolean  

If we do not understand data types properly,
we cannot clean data,
we cannot calculate correctly,
and we cannot build models properly.

# **Type Conversion**

Sometimes we need to convert one data type into another.

This is called **type conversion.**

In [ ]:
num1 = "10"
num2 = "20"

print(num1 + num2)

1020


The output will be **1020**.

Because both values are strings. (*Numbers are inside double quotes so python will identify it as string.*)

Python joins text. It does not add them as numbers.

In [ ]:
num1 = int(num1)
num2 = int(num2)

print(num1 + num2)
print(type(num1))

# We used int to convert num1 and num2 data type to integer. So now as it converted to interger python will add it like a calculator.

30
<class 'int'>


# **Practice Task:**

1. Create a variable called city and store your city name.
2. Create a variable called salary and store your monthly expenditure.
3. Print the type of both variables.
4. Store your age as a string.
5. Convert that age into integer.
6. Print the type before and after conversion.
7. What is the difference between 10 and "10"?
8. Why is understanding data types important in Data Science?
9. Imagine you are storing basic information about a customer. Create the following variables:

    * customer_name  
    * customer_age  
    * account_balance  
    * is_active  

    1. Print all the values.
    2. Print the data type of each variable.